apps/src/gamelab/animationListModule.js

```diff
@@ -164,6 +164,54 @@ function animationPropsReducer(state, action) {
 }
 
 /**
+ * Used to order numbers for sort()
+ */
+function compareNumbers(a, b) {
+  return a - b;
+}
+
+/**
+ * Given a name and an animationList, determine if the name is unique.
+ */
+export function isNameUnique(name, animationList) {
+  for (let animation in animationList) {
+    if (animationList[animation].name === name) {
+      return false;
+    }
+  }
+  return true;
+}
+
+/**
+ * Given a baseName and a animationList, provide a unique name
+ */
+function generateAnimationName(baseName, animationList) {
+  let unavailableNumbers = [];
+  for (let animation in animationList) {
+    let existingName = animationList[animation].name;
+    let lastUnderscore = existingName.lastIndexOf("_");
+    let lastDigits = existingName.substring(lastUnderscore + 1);
+    let existingBaseName = existingName;
+    if (/^\+?(0|[1-9]\d*)$/.test(lastDigits)) {
+      existingBaseName = existingName.substring(0, lastUnderscore);
+      if (existingBaseName === baseName) {
+        unavailableNumbers.push(parseInt(lastDigits));
+      }
+    }
+  }
+  unavailableNumbers.sort(compareNumbers);
```

Comment:
It's... really sad that this is necessary, but nice catch.  You could inline your sort function here.
```javascript
unavailableNumbers.sort((a, b) => a - b);
```